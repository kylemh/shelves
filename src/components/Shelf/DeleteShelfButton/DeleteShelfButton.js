import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styles from './DeleteShelfButton.module.scss';

export default class DeleteShelfButton extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    onClick: PropTypes.func.isRequired,
  };

  render() {
    const { children, onClick } = this.props;

    return (
      <button className={styles.DeleteShelfButton} onClick={onClick}>
        {children}
      </button>
    )
  }
}
