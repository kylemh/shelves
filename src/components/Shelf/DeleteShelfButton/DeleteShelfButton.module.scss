@import '../../../styles/variables.scss';

.DeleteShelfButton {
  background-color: inherit;
  border: 1px solid $white;
  color: $white;
  padding: 0.5rem;

  &:hover {
    cursor: pointer;
  }
}
