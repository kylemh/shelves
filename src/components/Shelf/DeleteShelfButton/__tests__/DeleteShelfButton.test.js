import React from 'react';
import { shallow } from 'enzyme';
import DeleteShelfButton from '../DeleteShelfButton';

describe('DeleteShelfButton', () => {
  it('should render', () => {
    const wrapper = shallow(<DeleteShelfButton onClick={jest.fn()}>ðŸ—‘</DeleteShelfButton>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call props.createShelf when clicked', () => {
    const mockDeleteShelfFunc = jest.fn();
    const wrapper = shallow(
      <DeleteShelfButton onClick={mockDeleteShelfFunc}>Trash Test</DeleteShelfButton>
    );

    expect(mockDeleteShelfFunc).toHaveBeenCalledTimes(0);
    wrapper.simulate('click');
    expect(mockDeleteShelfFunc).toHaveBeenCalledTimes(1);
  });
});
