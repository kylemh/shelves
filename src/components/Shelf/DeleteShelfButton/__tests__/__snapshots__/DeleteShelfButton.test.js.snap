// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteShelfButton should render 1`] = `
<button
  className="DeleteShelfButton"
  onClick={[MockFunction]}
>
  ðŸ—‘
</button>
`;
